dcp-0006/dcp-0006.mediawiki

```diff
@@ -0,0 +1,347 @@
+
+A treasury add transaction is identified by the following characteristics:
+* The wire message transaction version shall be 3.
+* There shall be at least one TxIn script.
+* There shall be one or two TxOut scripts.
+* All TxOut scripts shall be version 0.
+* TxOut[0] script must be <code>OP_TADD</code>.
+* TxOut[1] is optional if there is no change. If there is change it shall be a default <code>OP_SSTXCHANGE</code> script.
+
+{|
+!Transaction!!Script!!Description
+|-
+|TxIn[0..N]||Default utxo scripts||Normal utxo scripts as you would find on normal transactions.
+|-
+|TxOut[0]||[0]OP_TADD||Value shall be what is credited to the treasury.
+|-
+|TxOut[1]||[0]OP_SSTXCHANGE||Optional if there is no change.
+|}
+
+Additional consensus enforced characteristics:
+* A TAdd transaction shall not have a zero value for change.
+* There shall be no more than twenty TAdd transactions per block.
+
+===Debit treasury===
+
+The TSpend transactions are rather complex and some additional definitions need
+to be made prior to explaining the format.
+
+Required chain configuration parameters:
+* Treasury Vote Interval, or TVI, determines in which blocks a TSpend may occur. A block is considered on a TVI when the block height modulo TVI is equal to zero.
+* Treasury Vote Interval Multiplier, or TVIM, dictates the total duration (total duration = TVI * TVIM) where votes for or against the expenditure are considered valid.
+* Treasury Vote Quorum Multiplier and Divisor, TVQM, and TVQD respectively, are used to calculate the required number of votes to reach quorum.
+* Treasury Vote Required Multiplier and Divisor, TVRM and TVRD respectively, are used to calculate the required number of Yes votes.
+* Treasury Expenditure Policy, or TEP, determines the overall duration where all treasury expenditures are summed and averaged.
+* Treasury Expenditure Window, or TEW, is used to calculate the window that is used to verify that the sum of all TSpends does not exceed ~150% of the average treasury spending.
+* Treasury Expenditure Bootstrap, or TEB, is used to determine a baseline expenditure in order to bootstrap treasury payouts.
+* Politeia Keys, or PiKeys, are a number of well-known secp256k1 public keys that are allowed to sign TSpend transactions.
+
+Only the Pi key administrator can author transactions that debit the treasury
+account and generate utxos for recipients.
+
+The life-cycle of a TSpend transaction is as follows
+# PiKey administrator creates a TSpend transaction with an Expiry in the future.
+# The TSpend is signed and broadcast to the network.
+# If the TSpend is nominally valid (Expiry in the future, valid signature, valid public key etc) it is allowed in the mempool.
+# Once the TSpend hits the mempool wallets will be notified.
+# Voting wallets with a voting policy will now start voting either yes/no or abstain at the TSpend vote window start.
+# Votes are accumulated every block and once they reach an insurmountable tally they are short circuited.
```

Comment:
It is important to note that a vote can run shorter than the length of time it was allotted. This is fundamentally different from the consensus vote process. The important thing is that the only way a short-circuit can occur is when the remaining outstanding votes cannot alter the vote outcome anymore.

I'll take suggestions on how you think that would be easier to read.

Reply:
So if a voting interval is from N to N+100, wallets begin to vote at N. Lets say every vote is "yes", and this becomes insurmountable at N+50 - what actually happens at height N+50? I don't think the TSPEND is mined before N+100, so what does happen at N+50? Wallets stop voting?

Reply:
The tspend **is** mined before N+100. After block N+50 in your example, it will be mined on the next TVI block.

Reply:
The TSpend *is* mined on the next TVI if the vote tally is insurmountable.

Wallets should stop voting however they may not. The vote simply no longer counts since the outcome has been determined.

Reply:
Great, thanks for explaining. Perhaps this could be written out explicitly to avoid any confusion in future? Perhaps something like the following:

> Votes are accumulated every block. It is possible for a vote to be "short-circuited" if the outcome is already determined on a TVI block before the end of the voting interval. If an insurmountable tally of yes votes is reached, the TSpend is added to the TVI block. If an insurmountable tally of no votes is reached, the TSpend is removed from the mempool on the TVI block.

Reply:
That's fine. I changed it.